tests/ui/impl-trait/recursive-bound-eval.rs

```diff
@@ -17,8 +15,10 @@ impl<E, T: Fn() -> E> Parser<E> for T {
 }
 
 pub fn recursive_fn<E>() -> impl Parser<E> {
+    //[current]~^ ERROR: cycle
     move || recursive_fn().parse()
-    //[next]~^ ERROR: type annotations needed
+    //~^ ERROR: type annotations needed
+    //[current]~| ERROR: no method named `parse`
 }
```

Comment:
I guess we need to fix this issue for the new solver and the current one before we can land this commit